File: server/src/domain/measurement.ts
export const measurementTypes = ["weight", "pressure", "pulse"] as const;
export type MeasurementType = (typeof measurementTypes)[number];

export interface Measurement {
  id: string;
  type: MeasurementType;
  value: number;
  unit: string;
  measuredAt: string;
}

export type NewMeasurement = Omit<Measurement, "id">;

File: internal/domain/measurement.go
package domain

// MeasurementType identifies a supported health metric.
type MeasurementType string

const (
	MeasurementTypeWeight   MeasurementType = "weight"
	MeasurementTypePressure MeasurementType = "pressure"
	MeasurementTypePulse    MeasurementType = "pulse"
)

func (measurementType MeasurementType) Valid() bool {
	switch measurementType {
	case MeasurementTypeWeight, MeasurementTypePressure, MeasurementTypePulse:
		return true
	default:
		return false
	}
}

type Measurement struct {
	ID         string          `json:"id"`
	Type       MeasurementType `json:"type"`
	Value      float64         `json:"value"`
	Unit       string          `json:"unit"`
	MeasuredAt string          `json:"measuredAt"`
}

type NewMeasurement struct {
	Type       MeasurementType
	Value      float64
	Unit       string
	MeasuredAt string
}
